hipcheck/src/cache/plugin.rs

```
@@ -1,21 +1,154 @@
+	wd: Box<dyn Iterator<Item = StdResult<DirEntry, walkdir::Error>>>
+}
+
+impl HcPluginCacheIterator { 
+	fn new(path: &Path) -> Self {
+		HcPluginCacheIterator{
+			wd: Box::new(
+				WalkDir::new(path) //builds a new iterator using WalkDir
+					.min_depth(3) //makes sure to get the publisher, name, and version folders
+					.into_iter()
+					.filter_entry(|e| e.path().is_dir()), //makes sure the path is a directory
+			)
+		}	
+	}
+	fn path_to_plugin_entry(&self, path: &Path) -> Result<PluginCacheEntry> { //does function need to work when you clone a repo?
+	   let components: Vec<String> = path
+        .components()
+        .filter_map(|component| {
+            if let Component::Normal(name) = component { //extracts components seperated by "/" from the path
+                name.to_str().map(|s| s.to_string()) //converts the string slices into Strings
+            } else {
+                None
+            }
+
+        })
+        .collect(); //collects filepath into vector
+//Todo: add error handling for when the path length is less than 3
+		let relevant_components: &[String] = &components[components.len() - 3..];
+		let plugin_publisher = relevant_components[0].to_owned();
+		let plugin_name = relevant_components[1].to_owned();
+		let plugin_version = relevant_components[2].to_owned();
+		let last_modified: SystemTime = repo::get_last_modified_or_now(path);
+		Ok(PluginCacheEntry{
+							publisher:plugin_publisher, 
+							name:plugin_name, 
+							version: plugin_version,
+							modified: last_modified
+							}) 
+		
+	}
+	
+}
+
+
+
+///Function will take the full file path and put the last 3 directories (publisher, plugin, and review)
+///into the Plugin Cache entry
+impl Iterator for HcPluginCacheIterator {
+	type Item = PluginCacheEntry;
+	fn next(&mut self) -> Option<Self::Item> { 
+		if let Some(Ok(e)) = self.wd.next() {
+			if let Ok(ce) = self.path_to_plugin_entry(e.path()) {
+				Some(ce)
+			} 
+			else {
+				None
+			}
```

this `if else` structure can be replaced with doing `.ok()` on the output of `self.path_to_plugin_entry(e.path())`, which does the same thing. Turns `Result<T>` into `Option<T>`, discarding error. Same with the enclosing `if else`.